### PR TITLE
Add `validation_set()` for 3-way split approach

### PR DESCRIPTION
**Pull request by hfrick**
This is part 2 of introducing a 3-way split approach to the validation split: This PR adds a function `validation_set()` to make an `rset` from the result of `initial_validation_split()` and `group_initial_validation_split()`.

The accessor methods are currently as follows, with `val_split` being the class of the split objects inside the `rset`:

 accessor / object           | `initial_validation_split` | `val_split`
---            | ---                        | ---        
`training()`   | training                   | training   
`validation()` | validation                 | validation 
`testing()`    | testing                    | error      
`analysis()`   | error                      | training   
`assessment()` | error                      | validation 

For other types of `rsplit` objects inside of an `rset`, `training()` and `testing()` will error (see #394). I've made them work here because the split object isn't actually resampled (all the sampling happens in `initial_validation_split()`) but I'd like a second opinion here! We can also make `training()` and `validation()` error on `val_split` objects.

Similarly, do we want accessor functions for the `rset` returned by `validation_set()`? For other resamples, these do not exist but for this case, it sorta feels like they should? Or more precisely: should it work because it's just one split in the set? This would also apply to the other sets with only one split: `validation_split`, `validation_time_split`, `group_validation_split`, and `apparent`. I can also open a separate issue on this if we want to ponder this further.

### CODE REVIEW

File: R/validation_set.R [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#' Create a Validation Split for Tuning
+#'
+#' @param x An object of class `initial_validation_split`, such as resulting
```
Comment (hfrick) [inline R/validation_set.R:3]: What's a good name for this argument?

File: R/validation_set.R [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#' set.seed(1353)
+#' car_split <- initial_validation_split(mtcars)
+#' car_set <- validation_set(car_split)
+validation_set <- function(x, ...) {
+  rlang::check_dots_empty()
+
+  train_and_val <- rbind(
+    training(x),
+    validation(x)
+  )
```
Comment (hfrick) [inline R/validation_set.R:22]: This changes the order of rows compared to the initial data set going into `initial_validation_split()` - should it be the same order of rows?

File: R/validation_set.R (hunk 1 above)
Comment (topepo) [reply R/validation_set.R:3]: `split`?

File: R/validation_set.R (hunk 2 above)
Comment (topepo) [reply R/validation_set.R:22]: I don't think that it is a big deal.

File: R/validation_set.R [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#' @export
+validation.val_split <- function(x, ...) {
+  assessment(x, ...)
+}
+
+#' @rdname validation_set
+#' @export
+testing.val_split <- function(x, ...) {
+  rlang::abort(
+    "The testing data is not part of the validation set.", # TODO change error message?
```
Comment (topepo) [inline R/validation_set.R:74]:
```suggestion
    "The testing data is not part of the validation set object.", 
```

**Review by topepo — APPROVED** [review]

**Pull request merged**